### PR TITLE
[MRG] Example link

### PR DESCRIPTION
**Pull request by jaeilepp**
Takeover from https://github.com/mne-tools/mne-python/pull/3344

### CODE REVIEW

**Comment by jaeilepp** [comment]
Seems like it works now, but is very slow on circle. There is some issue on their site, so maybe I'll restart it later and see how it goes before we merge.

**Comment by jaeilepp** [comment]
Actually, now that I look at it, it looks like it was caching most of the time. Ready for review/merge.
Example here https://909-35097627-gh.circle-artifacts.com/0//home/ubuntu/mne-python/doc/_build/html/generated/mne.read_events.html?highlight=read_events#mne.read_events

**Comment by agramfort** [comment]
cool

any idea why we don't have the thumbnails and just the default helmet?

**Comment by jaeilepp** [comment]
That's just the circle build. It doesn't run the examples. We should see them when this is merged

**Comment by agramfort** [comment]
let's see !

File: doc/conf.py [hunk 1]
```diff
@@ -302,3 +302,14 @@
...
 
 numpydoc_class_members_toctree = False
+
+def touch_example_backreferences(app, what, name, obj, options, lines):
+    # generate empty examples files, so that we don't get
+    # inclusion errors if there are no examples for a class / module
+    examples_path = os.path.join(app.srcdir, "generated", "%s.examples" % name)
+    if not os.path.exists(examples_path):
+        # touch file
+        open(examples_path, 'w').close()
```
Comment (Eric89GXL) [inline doc/conf.py]: Why did you need to do this? If you are running latest `sphinx-gallery` it should not be necessary as they have lines like this in their code.

Comment (Eric89GXL) [inline doc/conf.py]: See related issue:

https://github.com/sphinx-gallery/sphinx-gallery/issues/126

And the fix that I committed that I think should make this unnecessary:

https://github.com/mne-tools/mne-python/commit/97b38c9a75c06ffaecf5eef588a689899f86ded2#diff-684a0ceb2878d48b19612ca360e277efR297

Comment (agramfort) [inline doc/conf.py]: I took these lines from nilearn

revert if needed.

Comment (Eric89GXL) [inline doc/conf.py]: What are they supposed to accomplish? Getting rid of build warnings? Those are gone already.

Comment (agramfort) [inline doc/conf.py]: whatever works

Comment (Eric89GXL) [inline doc/conf.py]: Okay, assuming that this is only to get rid of warnings we shouldn't need it, I'll commit to `master`
